### PR TITLE
Tuple support

### LINKED ISSUE #8: Tuple variant returns
As mentioned near the end of #4, it would be a good idea to allow for returning of tuples for tuple variants. For example (courtesy of @TheDan64),

```rust
enum Foo {
    Bar(bool, u32, f32),
}

fn into_bar(self) -> (bool, u32, f32);
fn as_bar(&self) -> &(bool, u32, f32);
// Or this one, but I think this may be equivalent to the one above
fn as_bar(&self) -> (&bool, &u32, &f32);
```

For `as_*` methods, the second variant should be returned, since we're referring to the values contained and not an actual tuple.

### PR DESCRIPTION
**Pull request by TheDan64**
Fixes #8

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -11,3 +14,33 @@ pub(crate) fn to_snake_case<S: AsRef<str>>(ident: &S) -> String {
...
+        UniqueIdentifierIterator {
+            buffer: vec!['a' as u8],
+        }
+    }
+}
+
+impl Iterator for UniqueIdentifierIterator {
+    type Item = Ident;
+
+    // Generates inifinite length strings from ASCII chars a-z
```
Comment (alekratz) [inline src/util.rs]: Go ahead and make this a rustdoc comment

Comment (alekratz) [inline src/util.rs]: Also, that's a really clever, generic way of making unique identifiers. I'm probably going to steal this in a few places 👍

Comment (TheDan64) [inline src/util.rs]: Thanks! And sure thing!

**Comment by alekratz** [comment]
Looks great. Thanks a bunch.

**Pull request merged**